app/workers/data_request_worker.rb

```diff
@@ -39,6 +39,7 @@ def perform(request_id)
       exporter.dump(path, estimated_count: estimated_count) do |progress, total|
         actual_count += 1
         if progress % 1000 == 0
+          request = DataRequest.find(request.id)
```

Comment:
I think this refresh db state lookup should be combined with the raise to fail as fast as possible
```suggestion
          if request.reload.canceling?
            raise DataRequest::DataRequestCanceled.new
          end
```